projects/ngx-datatable/src/lib/components/body/body.component.ts

```diff
@@ -866,8 +866,8 @@ export class DataTableBodyComponent<TRow extends Row = any> implements OnInit, O
     // If the detailRowHeight is auto --> only in case of non-virtualized scroll
     if (this.scrollbarV && this.virtualization) {
       const detailRowHeight = this.getDetailRowHeight(row) * (expanded ? -1 : 1);
-      // const idx = this.rowIndexes.get(row) || 0;
-      const idx = this.getRowIndex(row);
+      // This is hopefully only called with non-grouped rows. Otherwise, the heightCache fails.
+      const idx = this.getRowIndex(row).index;
```

Comment:
Considering your change in [L951](https://github.com/siemens/ngx-datatable/pull/111/files#diff-e6bb3cf05c09551bb2b499b01834eea67c18421f6ac5a18aa55d52c397f87640R951) which previously always returned `0` and hence let this part fail gracefully, wouldn't it make more sense to keep this graceful fallback and fallback with `0` here too?

```suggestion
      const idx = this.getRowIndex(row)?.index ?? 0;
```

Reply:
Makes sense, strict null checks would indeed fail here.